contracts/VotingEscrow.vy

```diff
@@ -718,28 +428,28 @@ def balanceOfAt(addr: address, _block: uint256) -> uint256:
         if _min >= _max:
             break
         _mid: uint256 = (_min + _max + 1) / 2
-        if self.user_point_history[addr][_mid].blk <= _block:
+        if self.user_point_history[addr][_mid].blk <= height:
             _min = _mid
         else:
             _max = _mid - 1
 
     upoint: Point = self.user_point_history[addr][_min]
 
     max_epoch: uint256 = self.epoch
-    _epoch: uint256 = self.find_block_epoch(_block, max_epoch)
-    point_0: Point = self.point_history[_epoch]
+    epoch: uint256 = self.find_block_epoch(height, max_epoch)
+    point_0: Point = self.point_history[epoch]
     d_block: uint256 = 0
     d_t: uint256 = 0
-    if _epoch < max_epoch:
-        point_1: Point = self.point_history[_epoch + 1]
+    if epoch < max_epoch:
+        point_1: Point = self.point_history[epoch + 1]
         d_block = point_1.blk - point_0.blk
         d_t = point_1.ts - point_0.ts
     else:
         d_block = block.number - point_0.blk
         d_t = block.timestamp - point_0.ts
     block_time: uint256 = point_0.ts
     if d_block != 0:
-        block_time += d_t * (_block - point_0.blk) / d_block
+        block_time += d_t * (height - point_0.blk) / d_block
 
     upoint.bias -= upoint.slope * convert(block_time - upoint.ts, int128)
```

Comment:
This requires some changes to keep max power for locks that are more than 4 years.

Reply:
`totalSupplyAt` should also be updated if this is changed.